lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -288,8 +305,16 @@ class WebMapServiceCatalogItem
 
   protected get defaultGetCapabilitiesUrl(): string | undefined {
     if (this.uri) {
-      return this.uri
-        .clone()
+      const baseUrl = QUERY_PARAMETERS_TO_REMOVE.reduce(
+        (url, parameter) =>
+          url
+            .removeQuery(parameter)
+            .removeQuery(parameter.toUpperCase())
+            .removeQuery(parameter.toLowerCase()),
+        this.uri.clone()
+      );
+
+      return baseUrl
         .setSearch({
           service: "WMS",
           version: this.useWmsVersion130 ? "1.3.0" : "1.1.1",
```

Comment:
Actually I see that version is hardcoded. IS this the best way to handle version? Should we allow people to define version themselves?

Reply:
The way we handle version is a bit convoluted - there are a few ways it can be defined by the user

- It is pulled out of `url` - which sets `useWmsVersion130`

https://github.com/TerriaJS/terriajs/blob/b979e4ddf09636de95f6b307690f1ee1dd3e3b75/lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts#L92-L99

- `useWmsVersion130` can be manually defined in the catalog JSON
- This is then used to determine version string - `this.useWmsVersion130 ? "1.3.0" : "1.1.1"`

`useWmsVersion130` is then used for the following:

- Different default parameters are then used for `GetMap` and `GetFeatureInfo` requests
https://github.com/TerriaJS/terriajs/blob/b979e4ddf09636de95f6b307690f1ee1dd3e3b75/lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts#L531-L542
- The version also determines whether `srs` or `crs` parameter is used
https://github.com/TerriaJS/terriajs/blob/b979e4ddf09636de95f6b307690f1ee1dd3e3b75/lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts#L578-L581

This is also not super robust - we don't handle WMS versions outside `1.3.0` and `1.1.1` - but so far it seems to work pretty well

Reply:
OK sounds like its well thought out!